Feature/send message with db

DebugMind opened the pull request:
## 📝 Pull Request 내용

### 📑 변경 사항
- [ ] 메세지 서비스 객체를 구현하였습니다.
  - 메세지를 전송하면 메세지가 message 테이블 에 저장됩니다.
  - 웹소켓이 끊어지면 chat_room_member 테이블 의 마지막으로 읽은 메세지의 id 가 갱신됩니다. 
- [ ] 웹소켓 인터셉터를 구현하여 웹소켓이 연결될 때 Security Context 에서 유저 id 를 받을 수 있도록 설정하였습니다.

### 🔗 관련 이슈
- #64 

### 💬 기타 참고 사항
- 테스트 코드를 짜볼려고 했는데, 웹소켓을 사용하는 부분이 있어서 쉽지 않더라구요..
  postman 에서 웹소켓을 연결해서 테스트를 해봤는데, 실시간 양방향 채팅이 잘되고 메세지도 잘 저장되며 웹소켓이 끊어질 때 마지막으로 읽은 메세지도 잘 갱신하는 것을 확인하였습니다 👍

DebugMind committed:
feat : message 서비스 별도로 분리

채팅방 메세지 서비스와 웹소켓을 별도로 분리하였습니다.

DebugMind committed:
feat : 채팅 메세지 전송 DTO 에 메세지 타입 추가

DebugMind committed:
feat : message 전송 시 DB 에 저장

DebugMind committed:
feat : websocket interceptor 추가

DebugMind committed:
refactor : class 명 변경 (ChatRoomWebSocketHandler -> ChatWebSocketHandler)

DebugMind committed:
feat : ChatMessageDto 스펙 수정

DebugMind committed:
feat : disconnect chat websocket

DebugMind committed:
feat : send Message

DebugMind committed:
feat : message 전송 DTO 수정

DebugMind committed:
feat : MessageService 에서 웹소켓 부분을 빼고 웹소켓 핸들러에 추가합니다.

lin-chae reviewed (approved):
정말 자세한 주석덕분에 이해가 정말 잘되네요. 많은 작업하시느라 고생하셨습니다 👍 👍 👍

DebugMind committed:
Merge remote-tracking branch 'origin/develop' into feature/send_message_with_DB

DebugMind committed:
feat : 앤티티 변화에 따른 시스템 로직 수정

chatRoomMember 에서 마지막으로 읽은 메세지의 id 를 저장하지 않고 앤티티 그 자체를 저장하도록 변경되었으므로 이에 맞게 수정하였습니다.

DebugMind commented:
chatRoomMember 에서 마지막으로 읽은 메세지의 id 를 저장하지 않고 앤티티 그 자체를 저장하도록 변경되었으므로 이에 맞게 수정하였습니다.

postman 으로 실시간 채팅 테스트 해보았는데, 실시간 양방향 채팅이 잘 되고, 메세지도 DB 에 잘 저장되며, 웹소켓이 끊겼을 때 마지막으로 읽은 메세지도 chat_room_member 테이블에 저장되는 것을 확인하였습니다 👍

djsumin commented on friends_server/src/main/kotlin/com/friends/chat/dto/ChatMessageDto.kt:
```diff
@@ -1,18 +1,18 @@
-    val senderProfileImageUrl: String,
     val message: String,
     val sendTime: LocalDateTime,
+    val type: MessageType,
```
메시지타입이라면 어떤식의 타입을 말씀하시는걸까요 ~~? 이미지인지 글인지 이걸 말씀하시는 건지 궁금합니다~

DebugMind replied:
넵 맞아여~
Message 앤티티에 있는 Enum 클래스인데 이미지, 글, 시스템 인지 구분해줍니다..!

djsumin reviewed (approved):
늦게 리뷰드려서 죄송합니다~😭 주석 덕분에 웹소켓 잘 모르는 저도 이해가 잘 갔습니다~~ 덕분에 많이 공부도 하고요!
